### ISSUE TITLE
Index Bug in geopandas/io/sql/read_postgis.py

### ISSUE THREAD

**Issue by henrymartin1** [geopandas/geopandas#996]
Original Code from read_postgis.py:
```Python
if crs is None:
        if len(geoms) > 0:
            srid = shapely.geos.lgeos.GEOSGetSRID(geoms[0]._geom)
            # if no defined SRID in geodatabase, returns SRID of 0
            if srid != 0:
                crs = {"init": "epsg:{}".format(srid)}
```

When infering the CRS, geopandas checks for the element in `geoms` with index = 0 instead of the first element. If 0 is not in the index (and the `index_col` is set) it is not possible to read the table.

Proposed fix:
Change `geoms[0]` to `geoms.iloc[0]` (line 60)

Proposition:
```Python
if crs is None:
        if len(geoms) > 0:
            srid = shapely.geos.lgeos.GEOSGetSRID(geoms.iloc[0]._geom)
            # if no defined SRID in geodatabase, returns SRID of 0
            if srid != 0:
                crs = {"init": "epsg:{}".format(srid)}```

**Comment by martinfleis** [comment]
Hi,
thanks for the report. However, you are apparently using older version of Geopandas, as this issue has been fixed in #856, which is part of 0.5.0 release. The affected line currently looks like this:

     srid = shapely.geos.lgeos.GEOSGetSRID(geoms.iat[0]._geom)

Try updating Geopandas. If the problem persists (as it should not based on your description), let us know. I will close this for now, as it is already fixed.

Thanks!

**Comment by henrymartin1** [comment]
Thanks for the quick answer! And sorry for not checking the version...
